jujuclient/mem.go

```diff
@@ -13,6 +15,8 @@ import (
 
 // MemStore is an in-memory implementation of ClientStore.
 type MemStore struct {
+	mu sync.Mutex
```

Comment:
http://reports.vapour.ws/releases/5260/job/run-unit-tests-race/attempt/2753

RemoveController() and AccountDetails() stomping on c.Accounts